### ISSUE TITLE
API: POST to /services/[SERVICE_NAME]/update Fails

### ISSUE THREAD

**Issue by afrazkhan** [moby/moby#29597]
**Description**

Sending a POST to the endpoint `/services/[SERVICE_NAME]/update` fails to update the service with a "service not found".

**Steps to reproduce the issue:**
e.g. reguest:
```
curl -X POST -H "Content-Type: application/json" --data-binary "@data.json"  -s --unix-socket /var/run/docker.sock http:/services/[SERVICE_NAME]/update?version=1
```

OR with [SHORT_SERVICE_ID]

```
curl -X POST -H "Content-Type: application/json" --data-binary "@data.json"  -s --unix-socket /var/run/docker.sock http:/services/[SHORT_SERVICE_ID]/update?version=1
```

**Describe the results you received:**
`{"message":"rpc error: code = 5 desc = service [SERVICE_NAME] not found"}`

**Describe the results you expected:**
`HTTP/1.1 200 OK` as documented [here](https://docs.docker.com/engine/reference/api/docker_remote_api_v1.24/#/update-a-swarm)

**Additional information you deem important (e.g. issue happens only occasionally):** Using the long service ID you can get from an inspect _does_ work

**Output of `docker version`:**
```
Client:
 Version:      1.12.5
 API version:  1.24
 Go version:   go1.6.4
 Git commit:   7392c3b
 Built:        Fri Dec 16 06:14:34 2016
 OS/Arch:      darwin/amd64

Server:
 Version:      1.12.5
 API version:  1.24
 Go version:   go1.6.4
 Git commit:   7392c3b
 Built:        Fri Dec 16 06:14:34 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 6
 Running: 1
 Paused: 0
 Stopped: 5
Images: 148
Server Version: 1.12.5
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 358
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: overlay bridge host null
Swarm: active
 NodeID: 4oh2xqqiexi4rk8julf5o9rj6
 Is Manager: true
 ClusterID: a6h0antslmorusyo7dx1z3fph
 Managers: 1
 Nodes: 1
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
 Node Address: 192.168.65.2
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 4.4.39-moby
Operating System: Alpine Linux v3.4
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 1.951 GiB
Name: moby
ID: KOMD:BA4Q:KMZP:LBEU:VFWV:MSPK:NDDL:REST:FJLU:TGIB:KVUU:GDOE
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 44
 Goroutines: 141
 System Time: 2016-12-20T14:25:35.166108835Z
 EventsListeners: 1
No Proxy: *.local, 169.254/16
Registry: https://index.docker.io/v1/
WARNING: No kernel memory limit support
Insecure Registries:
 0.0.0.0:5000
 127.0.0.0/8
```

**Comment by allencloud** [comment]
I am afraid it is because the code of: https://github.com/docker/docker/blob/v1.12.5/daemon/cluster/cluster.go#L921
Here dockerd uses the user input service name to pass to swarmkit side, while it should be the full ID.

while it has been fixed in docker/master: https://github.com/docker/docker/blob/master/daemon/cluster/cluster.go#L1046

**Comment by thaJeztah** [comment]
Looks like the API documentation for 1.24 is incorrect here indeed, as it was not implemented yet for 1.24 (on docker 1.12). @allencloud could you open a pull request to address that?

**Comment by allencloud** [comment]
Thanks for your remind. @thaJeztah 
PATL at pr https://github.com/docker/docker/pull/29600
